Fix Windows static build

neodesys opened the pull request:
On Windows, static compilation needs G_INTL_STATIC_COMPILATION
to be also defined during compilation. Else functions are exported with
__declspec(dllexport) which causes issues when importing symbols during
linkage with the static library.

xclaesse commented:
There was already `c_args += ['-DG_INTL_STATIC_COMPILATION']` a few lines down, shouldn't it be removed then?

xclaesse commented:
oh nevermind, c_args is for the declare_dependency, so it's actually in GLib that it shouldn't be needed: https://gitlab.gnome.org/GNOME/glib/-/blob/main/meson.build#L219.

Would be nice to get a release with this now that we can finally do static glib build on Windows :D

neodesys commented:
Yes, the `c_args += ['-DG_INTL_STATIC_COMPILATION']` is for the declare_dependency whereas what I added is to build the proxy-libintl library itself.

In fact under Linux or Mac it should have no effect as you don't have the cumbersome `dllimport/dllexport`. But under Windows you need both:
- the declare_dependency `c_args += ['-DG_INTL_STATIC_COMPILATION']`, else the importing project will try to use `dllimport` instead of `extern` for proxy-libintl symbols.
- what I've added to remove the `dllexport` while building the proxy-libintl static library. Else symbols are decorated with `dllexport` even if they are within a static library. The result is that when you link this static library, the upper project sees duplicated symbols and doesn't resolve then correctly.

neodesys commented:
What I added in the config.h of glib, it's just to enforce the fact that glib has been build with static proxy-libintl library because it can influence some of the imported headers. Just in case that somebody includes glib without using the meson dependencies system. But to be coherent I think it is useful to have this at both places, here and in glib.
